raster/src/main/scala/geotrellis/raster/DelegatingTile.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * This software is licensed under the Apache 2 license, quoted below.
+ *
+ * Copyright 2018 Astraea, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *     [http://www.apache.org/licenses/LICENSE-2.0]
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package geotrellis.raster
+
+/**
+ * A tile that wraps another tile. Originally intended for delayed reading, but useful in other special use cases.
+ *
+ * @since 8/22/18
+ */
+trait DelegatingTile extends Tile {
```

Comment:
What's the use case of having it in the `gt` codebase? Just wondering. Mb it makes sense to add some doc strings for it? If it's only useful for tests it makes sense to move it into the test folder.

Reply:
I wrote it for RasterFrames core, and have found it useful for a few use cases and thought it was general purpose enough for cases where users' might want to have a `Tile` with slightly different behavior by overriding selected methods. If it's in GT core I'll use that version in RF. If it's in `test` then I'll have to have my own copy of it.